File: services/auth/internal/domain/user_dto.go
package domain

import "time"

type UserDTO struct {
	ID          string  `json:"id"`
	Email       string  `json:"email"`
	Role        string  `json:"role"`
	IsAdmin     bool    `json:"is_admin"`
	CreatedAt   string  `json:"created_at"`
	DisplayName *string `json:"display_name,omitempty"`
}

func UserToDTO(u *User) UserDTO {
	dto := UserDTO{
		ID: u.ID.String(), Email: u.Email, Role: u.Role,
		IsAdmin: u.IsAdmin, CreatedAt: u.CreatedAt.UTC().Format(time.RFC3339),
	}
	if u.DisplayName != nil {
		dto.DisplayName = u.DisplayName
	}
	return dto
}

File: frontend/flutter/lib/api/models/user_dto.dart
/// Mirrors `UserDTO` from the backend (`DTO.md`).
///
/// ```
/// UserDTO {
///   id: string
///   email: string
///   display_name?: string
///   role: "student" | "teacher"
///   is_admin: boolean
///   created_at: string  // ISO8601
/// }
/// ```
class UserDto {
  final String id;
  final String email;
  final String? displayName;

  /// "student" | "teacher". Kept as a raw string to tolerate unknown values;
  /// use [isTeacher] / [isStudent] for checks.
  final String role;
  final bool isAdmin;
  final DateTime createdAt;

  const UserDto({
    required this.id,
    required this.email,
    required this.displayName,
    required this.role,
    required this.isAdmin,
    required this.createdAt,
  });

  bool get isStudent => role == 'student';
  bool get isTeacher => role == 'teacher';

  factory UserDto.fromJson(Map<String, dynamic> json) {
    return UserDto(
      id: json['id'] as String,
      email: json['email'] as String,
      displayName: json['display_name'] as String?,
      role: (json['role'] as String?) ?? 'student',
      isAdmin: (json['is_admin'] as bool?) ?? false,
      createdAt: DateTime.parse(json['created_at'] as String),
    );
  }

  Map<String, dynamic> toJson() => {
        'id': id,
        'email': email,
        'display_name': displayName,
        'role': role,
        'is_admin': isAdmin,
        'created_at': createdAt.toIso8601String(),
      };
}
